clang-tools-extra/clangd/ClangdServer.cpp

```diff
@@ -196,7 +196,7 @@ void ClangdServer::_3CCollectAndBuildInitialConstraints(
     _3CDiagInfo.clearAllDiags();
     ConvCB->send3CMessage("Running 3C for first time.");
     _3CInter.buildInitialConstraints();
-    _3CInter.solveConstraints(true);
+    _3CInter.solveConstraints();
```

Comment:
Do you know if this actually works at runtime, or does it just fix the compile error?  I'm guessing it doesn't work because the argument previously controlled whether `computeInterimConstraintState` is called, but now the call is conditional on `WarnRootCause`, which AFAIK clangd3c doesn't set.  So if you just remove the argument, the call won't happen, and as a result, clangd3c probably won't have the information it needs.  It might be good to add a comment that this is one of the things we should check if we want to try to get clangd3c working again.  Or maybe clangd3c is so broken at runtime already that it's not worth tracking this.

I don't know what is the best way to handle clangd3c going forward.  As long as the code exists in our source tree, it seems that you and I are both going to be tempted to maintain it in some way (at least it feels to me like a pity not to include it in renames), but without actually being able to run it, it will be hard to determine what counts as appropriate maintenance and we're liable to just waste a lot of effort.

Reply:
If we're going to punt the general decision on clangd3c as John said, let me do the safe thing and make a suggestion for a comment:
```suggestion
    // NOTE: If and when we revive clangd3c, revisit this to make sure
    // computeInterimConstraintState gets called if clangd3c needs it.
    _3CInter.solveConstraints();
```

Reply:
Yes, at least leaving this comment seems prudent